tensorflow/core/kernels/mkl/mkl_pooling_ops_common.cc

```diff
@@ -54,8 +54,13 @@ void MklPoolingFwdPrimitive<T>::Setup(const MklPoolingParams& fwdParams) {
 #else
   context_.src_md.reset(new memory::desc(fwdParams.src_md.data));
 #endif  //  !ENABLE_MKLDNN_V1
-  context_.dst_md.reset(new memory::desc({fwdParams.dst_dims}, MklDnnType<T>(),
-                                         MEMORY_FORMAT::any));
+  if (!fwdParams.native_format) {
+    context_.dst_md.reset(new memory::desc(
+        {fwdParams.dst_dims}, MklDnnType<T>(), MEMORY_FORMAT::any));
+  } else {
+    context_.dst_md.reset(new memory::desc(
+        {fwdParams.dst_dims}, MklDnnType<T>(), fwdParams.src_format));
+  }
```

Comment:
Nit: I think this is easier to read:
```suggestion
  context_.dst_md.reset(new memory::desc(
      {fwdParams.dst_dims}, MklDnnType<T>(),
      fwdParams.native_format ? fwdParams.src_format : MEMORY_FORMAT::any));
```

Reply:
Done.